### PR TITLE
fix: [PL-26457]: Changing - to _ in scimUserGroupIdentifier

### PR DESCRIPTION
**Pull request by Y0709**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35519)

### CODE REVIEW

**Comment by Y0709** [comment]
trigger messagecheck

**Comment by abhinav-harness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/scim/NGScimGroupServiceImpl.java` line 364 at r1](https://reviewable.io/reviews/harness/harness-core/35519#-N7l2m25B-1I0zfcvmKH:-N7l2m25B-1I0zfcvmKI:bcubz7w) ([raw file](https://github.com/harness/harness-core/blob/fcc00fe9ca7cd0043c942b83dda21e2632c20bb9/120-ng-manager/src/main/java/io/harness/ng/scim/NGScimGroupServiceImpl.java#L364)):*
> ```Java
>         groupQuery);
>     String userGroupIdentifier = isNotEmpty(groupQuery.getDisplayName())
>         ? groupQuery.getDisplayName().replaceAll("\\.", "_").replaceAll("\\-", "_")
> ```

do we need escape charater for "-" ?

**Comment by abhinav-harness** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/scim/NGScimGroupServiceImpl.java` line 364 at r1](https://reviewable.io/reviews/harness/harness-core/35519#-N7l2qTy9JStsJHxZR5-:-N7l2qTy9JStsJHxZR50:b-41naml) ([raw file](https://github.com/harness/harness-core/blob/fcc00fe9ca7cd0043c942b83dda21e2632c20bb9/120-ng-manager/src/main/java/io/harness/ng/scim/NGScimGroupServiceImpl.java#L364)):*
> ```Java
>         groupQuery);
>     String userGroupIdentifier = isNotEmpty(groupQuery.getDisplayName())
>         ? groupQuery.getDisplayName().replaceAll("\\.", "_").replaceAll("\\-", "_")
> ```

what happens to existing records?

File: 120-ng-manager/src/test/java/io/harness/ng/scim/NGScimGroupServiceImplTest.java [hunk 1]
```diff
@@ -176,4 +177,89 @@ public void testSearchGroupByNameNoSkipNoCountReturnsDefaultResult() {
...
+  public void test_createUserGroupForDash2() {
+    String accountId = "accountId";
+
+    ScimGroup scimGroup = new ScimGroup();
+    scimGroup.setDisplayName("display-name");
+    scimGroup.setId("id");
+
+    UserGroup userGroup = UserGroup.builder()
+                              .name(scimGroup.getDisplayName())
+                              .identifier(scimGroup.getDisplayName().replaceAll("-", "_"))
```
Comment (abhinav-harness) [inline 120-ng-manager/src/test/java/io/harness/ng/scim/NGScimGroupServiceImplTest.java:215]: scimGroup.getDisplayName().replaceAll("-", "_")

make a variable for this

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35519)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35519)* status: :shipit: complete! ✓

**Pull request merged**